src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -154,72 +181,162 @@ export default class SimidMapCreative extends BaseSimidCreative {
+    };
+    const placeMarker = new google.maps.Marker({
+      map: this.map_,
+      position: place.geometry.location,
+      icon: placeIcon
+    });
+    ///Recalculate directions if new marker is clicked.
+    placeMarker.addListener('click', () => {
+      this.activeLocation_ = place.geometry.location;
+      this.displayDirections_(this.currentLocation_, this.activeLocation_);
+    });
+  }
+
+  /**
+   * Displays the route between the starting loaction and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} destination The LatLng coordinates of the destination.
+   * @param {!google.maps.LatLng} startingLocation The LatLng coordinates of the start location.
+   * @private 
+   */
+  displayDirections_(destination, startingLocation) {
+    this.directionsRenderer_.setMap(this.map_);
+    this.calculateRoute_(startingLocation, destination);
+    //If travel method changes, recalculate directions.
+    document.getElementById("travel_method").addEventListener("change", () => {
+      this.calculateRoute_(startingLocation, destination);
+    });
+  }
+
+  /**
+   * Creates a drop down menu where users can choose between
+   * different modes of travel to display directions for.
+   * @private 
+   */
+  createTravelChoices_() {
+    const travelChoicesContainer = document.getElementById("button_container")
+    const travelMethod = document.createElement('select');
+    travelMethod.id = "travel_method";
+    for(let i = 0; i < this.transportMethods_.length; i++) {
+      travelMethod.add(this.transportMethods_[i]);
+    }
+    travelMethod.classList.add("travel_method");
+    travelChoicesContainer.append(travelMethod);
+  }
+
+  /**
+   * Creates an option element representing a mode of travel.
+   * @param {!string} travelMode A string representing the
+   *   given mode of travel.
+   * @private 
+   */
+  createTravelOption_(travelMode) {
+    const travelOption = document.createElement("option");
+    travelOption.value = travelMode.toUpperCase();
+    travelOption.text = travelMode;
+    return travelOption;
+  }
+
+  /**
+   * Displays the route between the starting location and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} start The LatLng coordinates of the start location.
+   * @param {!google.maps.LatLng} end The LatLng coordinates of the end location.
+   * @private 
+   */
+  calculateRoute_(start, end) {
+    const directionsService = new google.maps.DirectionsService();
+    const selectedMode = document.getElementById("travel_method").value;
+    directionsService.route(
+      {
+        origin: start,
+        destination: end,
+        travelMode: [selectedMode]
+      },
+      (response, status) => {
+        if (status == "OK") {
+          this.directionsRenderer_.setDirections(response);
+        } else {
+          const directionsErrorMessage = {
+            message: "ERROR: Directions request failed due to " + status,
```

Comment:
does the map status provide enough information as to why it failed? Not completely sure how descriptive it can be for this sentence.

Reply:
'ERROR: Failed to load directions: + status"

Reply:
Made this adjustment